File: Toggle_String_Case/Toggle.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 13/08/2023
    Program : Write a program which accept string from user and toggle the case.
              
              Input : "Marvellous Multi OS"
              Output : mARVELLOUS mULTI os
*/

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
package main
import (
	"fmt"
	"bufio"
	"os"
)

//Function To Toggle Case
func Toggle(arr []byte) []byte {
	var i int = 0;

	//Logic To Toggle Case
	for i != len(arr) {
		if ((arr[i] >= 'A')&&(arr[i] <= 'Z')) {
			arr[i] = arr[i] + 32;
		} else if ((arr[i] >= 'a')&&(arr[i] <= 'z')) {
			arr[i] = arr[i] - 32;
		}

		i++;
	}

	return arr;
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var str string = " ";
	var arr []byte;
	var err error = nil;
	var sret []byte ;

	fmt.Println("Jay Ganesh....");

	//Taking input from user
	fmt.Println("Enter One String");
	var bobj = bufio.NewReader(os.Stdin);

	//Reading String 
	str,err = bobj.ReadString('\n');

	if (err != nil) {
		return;
	}

	arr = []byte(str);

	//Call To Toggle Function
	sret = Toggle(arr);

	//Priting Modified String 
	fmt.Printf("Modified String is : %s",sret);
}

File: Toggle_First_Last_Nibble/Toggle.ts
/*
    Author : Pakshal Shashikant Jain 
    Date : 07/09/2023
    Program : Write a program which accept one number from user and toggle contents 
              of first and last nibble of the number. Return modified number. (Nibble is a
              group of four bits) 

*/

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Nibble 
{
    //Characteristics of Class
    iNo : bigint;

    //Constructor of Class
    constructor()
    {
        this.iNo;
    }

    //Behaviour of Class To Take Input from user
    Accept()
    {
        this.iNo = 508n;
    }

    //Behaviour of Class To Toggle Content of First and Last Nibble of Entered Number
    Toggle() : bigint
    {
        var iMask : bigint = 0XF000000Fn;
        var iResult : bigint = 0n;

        if(this.iNo < 0)
        {
            this.iNo = -this.iNo;
        }

        //Logic To Toggle Contents
        iResult = iMask ^ this.iNo;

        return iResult;
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
function main() 
{
    var iret : bigint;

    console.log("Jay Ganesh....");

    //Creating object of Nibble Class
    var nobj = new Nibble();

    //Call To Behaviours of Class
    nobj.Accept();
    iret = nobj.Toggle();   

    //Printing Modified Number
    console.log("Modified Number is : ",iret);
}

main();